src/azure_devtools/scenario_tests/recording_processors.py

```diff
@@ -138,13 +141,14 @@ def __init__(self, replacement='fake_token'):
         self._replacement = replacement
 
     def process_response(self, response):
-        import json
-        try:
-            body = json.loads(response['body']['string'])
-            body['access_token'] = self._replacement
-        except (KeyError, ValueError):
-            return response
-        response['body']['string'] = json.dumps(body)
+        if is_text_payload(response):
```

Comment:
Not required here, since we catch ValueError if it's not JSON. `is_text_payload` is not enough otherwise, since it could be XML (for instance), so I would recommend just let the try/except do its job

Reply:
You are right. The exception I got is UnicodeDecodeError, which is a ValueError, so no need for the extra check